File: CS-Examples/24_Workbook/RemoveCustomProperties.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core;

namespace RemoveCustomProperties
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load a excel document
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\templateAz.xlsx");

            // Retrieve a list of all custom document properties of the Excel file
            ICustomDocumentProperties customDocumentProperties = workbook.CustomDocumentProperties;

            // Remove "Editor" custom document property
            customDocumentProperties.Remove("Editor");

            // Save to file
            String result = "RemoveCustomProperties_result.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources 
            workbook.Dispose();

            // Launch the document
            ExcelDocViewer(result);
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/24_Workbook/RemoveCustomProperties.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core

Namespace RemoveCustomProperties
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\templateAz.xlsx")

            ' Get the custom document properties of the workbook
            Dim customDocumentProperties As ICustomDocumentProperties = workbook.CustomDocumentProperties

            ' Remove the custom document property named "Editor"
            customDocumentProperties.Remove("Editor")

            ' Specify the file name for the resulting workbook to be saved
            Dim result As String = "RemoveCustomProperties_result.xlsx"

            ' Save the modified workbook to a file with the specified name and Excel version (Version2010)
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
